tests/utils/test_url.py

```diff
@@ -0,0 +1,118 @@
+from unittest.mock import MagicMock, patch
+
+import pytest
+from page_analyzer.utils.url import (
+    check_url,
+    normalize,
+    parse_page_data,
+    validate,
+)
+
+
+def test_validate():
+    # given
+    url = "https://www.weforum.org/my-forum/"
+
+    # when
+    errors = validate(url)
+
+    # then
+    assert errors == {}
+
+
+def test_validate_invalid():
+    # given
+    url = "htttps://www.weforum.org/my-forum/"
+
+    # when
+    errors = validate(url)
+
+    # then
+    assert errors == {"name": "Некорректный URL"}
+
+
+def test_normalize():
+    # given
+    url = "http://www.example.com:1030/software/index.html"
+
+    # when
+    normalized_url = normalize(url)
+
+    # then
+    assert normalized_url == "http://www.example.com"
+
+
+def test_check_url():
+    # given
+    url = "https://example.com"
+    parsing_data = {
+        "response_code": "200",
+        "h1": "some header",
+        "title": "some title",
+        "description": "some description",
+    }
+
+    # when
+    with patch("page_analyzer.utils.url.parse_page_data") as mock_parse_page_data:
+        with patch("page_analyzer.utils.url.requests") as mock_requests:
```

Comment:
if you want to use two patches in one place (or two context managers in one place), you can do this:

```suggestion
    with patch("page_analyzer.utils.url.parse_page_data") as mock_parse_page_data, patch("page_analyzer.utils.url.requests") as mock_requests:
```